app/controllers/assessments_controller.rb

```diff
@@ -117,16 +117,19 @@ def create
 
       # Determine if a user created this assessment or a monitoree
       @assessment.who_reported = current_user.nil? ? 'Monitoree' : current_user.email
-      @assessment.save
 
-      # Save a new receipt and clear out any older ones
-      AssessmentReceipt.where(submission_token: submission_token_from_params).delete_all
-      @assessment_receipt = AssessmentReceipt.new(submission_token: submission_token_from_params)
-      @assessment_receipt.save
+      reported_condition.transaction do
+        @assessment.save!
+        reported_condition.save!
 
-      # Create history if assessment was created by user
-      History.report_created(patient: patient, created_by: current_user.email, comment: "User created a new report (ID: #{@assessment.id}).") if current_user
+        # Save a new receipt and clear out any older ones
+        AssessmentReceipt.where(submission_token: submission_token_from_params).delete_all
+        @assessment_receipt = AssessmentReceipt.new(submission_token: submission_token_from_params)
+        @assessment_receipt.save
```

Comment:
I'm thinking we might actually want this to be a `save!`. If not, we might want to at least check if it saves properly before creating the History item below. I know that was existing functionality, but I figure, while we're here and trying to prevent Assessment saving issues would be nice to add.

Reply:
Even if this is an `AssessmentReceipt` rather than an `Assessment`?

Reply:
Might also be worth seeing if we can actually consolidate these three lines using ActiveRecord upsert, which was just added in Rails 6

Reply:
I guess if the Assessment was successfully saved but the Assessment receipt wasn't, it's still okay to make the History item, so maybe not then.